ibeam/src/gateway_client.py

```diff
@@ -173,7 +191,54 @@ def authenticate_gateway(driver, account, password, key: str = None, base_url: s
         submit_form_el.click()
 
         success_present = EC.text_to_be_present_in_element((By.TAG_NAME, 'pre'), _SUCCESS_EL_TEXT)
-        WebDriverWait(driver, _OAUTH_TIMEOUT).until(success_present)
+        two_factor_input_present = EC.presence_of_element_located((By.ID, _TWO_FAC_EL_ID))
+
+        WebDriverWait(driver, _OAUTH_TIMEOUT).until(AnyEc(success_present, two_factor_input_present))
+
+        two_factor_el = driver.find_elements_by_id(_TWO_FAC_EL_ID)
+
+        if two_factor_el:
```

Comment:
Also, we'll need to tweak this logic. The 'chlginput' element is visible on the page all the time, therefore checking for it does not indicate whether 2FA was requested. Any idea what else changes when 2FA is requested? I can't reproduce it on my end.

Reply:
Ah okay we can do this I think
```if two_factor_el.isDisplayed():```

Reply:
Nice suggestion!

And how could we go over this bit?
```python
        two_factor_input_present = EC.presence_of_element_located((By.ID, var.TWO_FAC_EL_ID))

        WebDriverWait(driver, var.OAUTH_TIMEOUT).until(AnyEc(success_present, two_factor_input_present))
```
Here it is checking for presence of `chlginput` which is present at that point already. Ideas?

Reply:
Right, I think I got it: element of id `twofactbase`

Reply:
Ok, got this sorted 👍